rdflib/namespace/__init__.py

```diff
@@ -590,6 +590,31 @@ def compute_qname_strict(
 
             return self.__cache_strict[uri]
 
+    def expand_curie(self, curie: str) -> Union[URIRef, None]:
+        """
+        Expand a qname (aka CURIE) of the form <prefix:element>, e.g. "rdf:type"
+        into its full expression:
+
+        >>> import rdflib
+        >>> g = rdflib.Graph()
+        >>> g.namespace_manager.expand_curie("rdf:type")
+        'http://www.w3.org/1999/02/22-rdf-syntax-ns#type'
+
+        Returns None if a namespace is not bound to the prefix.
+
+        """
+        if not type(curie) is str:
+            raise TypeError("Argument must be a string.")
```

Comment:
I'm understand somewhat why you are doing this, to raise an error if someone accidentally passes in a  URIRef, but I'm not sure it is worth it, because this will then break other subclasses of string also. Would like to hear other opinions though.

Reply:
Maybe the best compromise here is to rather do:

```suggestion
        if not isinstance(curie, str) or isinstance(curie, Identifier):
            raise TypeError("Argument must be a string but should not be an `rdflib.term.Identifier`")
```

That way if someone passes in something like a URIRef, BNode or Literal accidentally they will get an error. It may still be going too far to try and save someone, those are strings, and if this takes a string I think it may be best to just accept that if someone does it they mean to do it.

Reply:
I did start with `isinstance(curie, str)` but on testing, that proved too lax a check, hence the use of the more restrictive `type` and specific admonition to use a string. But making the error message more informative is worth doing and I added some accompanying tests and yes, in the case of an unbound prefix, it should return an informative exception rather than just `None`.